x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -1216,7 +1217,36 @@ private Expression implicitCast(Expression e) {
         }
     }
 
-    public static class PruneSubqueryAliases extends AnalyzerRule<SubQueryAlias> {
+    public static class ReplaceSubQueryAliases extends AnalyzerRule<UnaryPlan> {
+
+        @Override
+        protected LogicalPlan rule(UnaryPlan plan) {
+            if (plan.child() instanceof SubQueryAlias) {
+                SubQueryAlias a = (SubQueryAlias) plan.child();
+                return plan.transformExpressionsDown(FieldAttribute.class, f -> {
+                   if (f.qualifier() != null && f.qualifier().equals(a.alias())) {
+                       // Find the underlying concrete relation (EsIndex) and its name as the new qualifier
+                       String newQualifier = null;
+                       List<LogicalPlan> children = a.collectFirstChildren(p -> p instanceof EsRelation);
+                       if (children.isEmpty() == false) {
+                           newQualifier = ((EsRelation) children.get(0)).index().name();
+                       }
```

Comment:
Shouldn't an `else` just return `f` here? Otherwise it seems to strip the qualifier.
Not sure if it can happen (maybe when ref'ing an `UnresolvedRelation`?), but might be safer.

Reply:
Yep, makes sense, It shouldn't happened, but I fixed it to return f instead of stripping completely the qualifier which could hide some other issue in the future.